tamr_toolbox/notifications/emails.py

```diff
@@ -0,0 +1,212 @@
+import ssl
+
+from typing import Union, List, Optional, Tuple
+from tamr_toolbox.models.data_type import JsonDict
+from email.mime.text import MIMEText
+from tamr_toolbox.notifications.common import _monitor_job as monitor_job_common
+
+from tamr_unify_client import Client
+from tamr_unify_client.operation import Operation
+
+from tamr_toolbox.models.operation_state import OperationState
+from tamr_toolbox.utils.operation import get_details
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _build_message(*, message: str, subject_line: str, sender: str, recipients: List[str]) -> str:
+    """Builds email message in Multipurpose Internet Mail Extensions (MIME) format. MIME is an
+    Internet standard that extends the format of email messages
+
+    Args:
+        message: Body of email message
+        subject_line: subject of email
+        sender: email address of sender
+        recipients: list of emails to send message to
+
+    Returns:
+        Email as a string
+    """
+
+    # build email
+    msg = MIMEText(message)
+    msg["Subject"] = subject_line
+    msg["From"] = sender
+    msg["To"] = ", ".join(recipients)
+    return msg.as_string()
+
+
+def send_email(
+    *,
+    message: str,
+    subject_line: str,
+    sender_address: str,
+    sender_password: str,
+    recipient_addresses: List[str],
+    smtp_server: str,
+    smtp_port: str,
+    use_tls: bool = True,
+    keyfile: Optional[str] = None,
+    certfile: Optional[str] = None,
+) -> JsonDict:
+    """Sends a message via email to list of recipients
+
+    Args:
+        message: Body of email message
+        subject_line: subject of email
+        sender_address: email address to send message from ex: my_pipeline@gmail.com
+        sender_password: password to login to sender_email
+        recipient_addresses: list of emails to send message to ex: [client_email@gmail.com]
+        smtp_server: smtp server address of sender_email ex: smtp.gmail.com
+        smtp_port: port to send email from, use 465 for SSL, use 587 for TLS
+        use_tls: A boolean value to turn on/off TLS protocol
+        keyfile: the private key to a TLS/SSL certificate, usually PEM format
+        certfile: TLS/SSL cert file issued by a Certificate Authority (CA), usually PEM format
+
+    Returns:
+        A dict with the message and response codes from the smtp server if there are any
+        errors. The dictionary will contain one entry for each recipient that was refused.
+        Each entry contains a tuple of the SMTP error code and the accompanying error
+        message sent by the server. A successful response will contain an empty dict.
+
+    Raises:
+        SMTPException. The base exception class used by the smtplib module
+    """
+    # build email
+    msg = _build_message(
+        message=message,
+        subject_line=subject_line,
+        sender=sender_address,
+        recipients=recipient_addresses,
+    )
+    response = dict()
+    response["message"] = message
```

Comment:
Nit: This can be a little bit more concise, but really not an issue at all
```suggestion
    response = {"message": message}
```